perf(ast): use bitflags for storing comment newline state

camchenry opened the pull request:
- supersedes https://github.com/oxc-project/oxc/pull/11090

This switches the newline flags to using `bitflags` to make the setting of the flags a little bit simpler and also makes checking the flag state easier too. Rather than using all flags in a bitflags, this is simpler because we don't need a custom serializer for comments for ESTree.

camchenry commented:
* **#11096** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11096?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/11096?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=11096).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

overlookmotel reviewed (approved):
Great. Thank you for persisting, despite my misdirection (twice!)

overlookmotel commented:
### Merge activity

* **May 17, 7:42 PM EDT**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

camchenry committed:
perf(ast): use bitflags for storing comment newline state (#11096)

- supersedes https://github.com/oxc-project/oxc/pull/11090

This switches the newline flags to using `bitflags` to make the setting of the flags a little bit simpler and also makes checking the flag state easier too. Rather than using all flags in a bitflags, this is simpler because we don't need a custom serializer for comments for ESTree.
